src/devices/Bmxx80/Bmxx80Base.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.IO;
+using Iot.Device.Bmxx80.CalibrationData;
+using Iot.Device.Bmxx80.PowerMode;
+using Iot.Device.Bmxx80.Register;
+using Iot.Units;
+
+namespace Iot.Device.Bmxx80
+{
+    /// <summary>
+    /// Represents the core functionality of the Bmxx80 family.
+    /// </summary>
+    public abstract class Bmxx80Base : IDisposable
+    {
+        internal Bmxx80CalibrationData _calibrationData;
```

Comment:
Should all of these internal fields be protected instead? That way, we can have people define their own BME980 (for example) binding in their own project in the future and still use the logic in this base class.